src/webgpu/api/validation/queue/copyImageBitmapToTexture.spec.ts

```diff
@@ -0,0 +1,409 @@
+  ): GPUTextureCopyView {
+    // In generated tests, there are some pairs will trigger validation errors here, eg. sampleCount
+    // 4 and BC formats. Filter these validation and let copyImageBitmapToTexture to catch them.
+    this.device.pushErrorScope('validation');
+    const dstTexture = this.device.createTexture({
+      size,
+      sampleCount,
+      dimension,
+      mipLevelCount: DEFAULT_MIPLEVEL_COUNT,
+      format,
+      usage,
+    });
+
+    if (destroyed) {
+      dstTexture.destroy();
+    }
+    this.device.popErrorScope();
+
+    return { texture: dstTexture, mipLevel, origin };
+  }
+
+  // Generate dst texture copy view with default width, height.
+  getTextureCopyViewWithDefaultSize(
+    sampleCount: number,
+    dimension: GPUTextureDimension,
+    format: GPUTextureFormat,
+    usage: GPUTextureUsageFlags,
+    mipLevel: number,
+    origin: GPUOrigin3D,
+    destroyed: boolean
+  ): GPUTextureCopyView {
+    let size: GPUExtent3D = { width: 0, height: 0, depth: 0 };
+    switch (dimension) {
+      case '1d':
+        size = { width: DEFAULT_WIDTH, height: 1, depth: 1 };
+        break;
+      case '2d':
+        size = { width: DEFAULT_WIDTH, height: DEFAULT_HEIGHT, depth: 1 };
+        break;
+      case '3d':
+        size = { width: DEFAULT_WIDTH, height: DEFAULT_HEIGHT, depth: DEFAULT_DEPTH };
+        break;
+      default:
+        unreachable('unknown texture dimension');
+    }
+
+    return this.getTextureCopyView(
+      size,
+      sampleCount,
+      dimension,
+      format,
+      usage,
+      mipLevel,
+      origin,
+      destroyed
+    );
+  }
+
+  // Generate dst texture copy view with default width, height and correct configs.
+  getDefaultTextureCopyView(dimension: GPUTextureDimension): GPUTextureCopyView {
+    return this.getTextureCopyViewWithDefaultSize(
+      1,
+      dimension,
+      'bgra8unorm',
+      GPUTextureUsage.COPY_DST,
+      0,
+      { x: 0, y: 0, z: 0 },
+      false
+    );
+  }
+
+  testRun(
+    imageBitmapCopyView: GPUImageBitmapCopyView,
+    textureCopyView: GPUTextureCopyView,
+    copySize: GPUExtent3D,
+    success: boolean,
+    exceptionName?: string
+  ): void {
+    // CopyImageBitmapToTexture will generate two types of errors. One is exceptions generated out
+    // of Dawn and the other is validation errors from Dawn.
+    if (exceptionName && exceptionName.length !== 0) {
+      this.shouldThrow(exceptionName, () => {
+        this.device.defaultQueue.copyImageBitmapToTexture(
+          imageBitmapCopyView,
+          textureCopyView,
+          copySize
+        );
+      });
+    } else {
+      this.expectValidationError(() => {
+        this.device.defaultQueue.copyImageBitmapToTexture(
+          imageBitmapCopyView,
+          textureCopyView,
+          copySize
+        );
+      }, !success);
+    }
+  }
+}
+
+export const g = makeTestGroup(CopyImageBitmapToTextureTest);
+
+g.test('copyImageBitmapToTexture_source_imageBitmap_copy_view')
```

Comment:
nit: copyImageBitmapToTexture is redundant here, because it's already in the filename. Same for the rest of the tests.

Reply:
Done.